flow/layers/image_filter_layer.cc

```diff
@@ -10,12 +10,12 @@
 
 namespace flutter {
 
-ImageFilterLayer::ImageFilterLayer(std::shared_ptr<const DlImageFilter> filter,
+ImageFilterLayer::ImageFilterLayer(const std::shared_ptr<DlImageFilter>& filter,
```

Comment:
I don't think switching these to const references is an improvement.  With the std::move there ways to avoid a copy, the const reference forces a copy to happen.

Reply:
These are always called from a reference source that cannot be moved so a copy must happen at some point. It would be better if the copy happened into the layer's local field rather than to the value handed in to the constructor.

Reply:
You are considering just the code that is written today.  There is no way to enforce that this is only called with references.